src/rpp/rpp/operators/on_error_resume_next.hpp

```diff
@@ -0,0 +1,109 @@
+//                  ReactivePlusPlus library
+//
+//          Copyright Aleksey Loginov 2023 - present.
+// Distributed under the Boost Software License, Version 1.0.
+//    (See accompanying file LICENSE_1_0.txt or copy at
+//          https://www.boost.org/LICENSE_1_0.txt)
+//
+// Project home: https://github.com/victimsnino/ReactivePlusPlus
+//
+
+#pragma once
+
+#include <rpp/operators/fwd.hpp>
+
+#include <rpp/defs.hpp>
+#include <rpp/operators/details/strategy.hpp>
+
+namespace rpp::operators::details
+{
+    template<rpp::constraint::observer TObserver, rpp::constraint::decayed_type Selector>
+    struct on_error_resume_next_observer_strategy
+    {
+        using preferred_disposable_strategy = rpp::details::observers::none_disposable_strategy;
+
+        mutable TObserver observer;
+        Selector          selector;
+        // Manually control disposable to ensure observer is not used after move in on_error emission
+        mutable rpp::composite_disposable_wrapper disposable = composite_disposable_wrapper::make();
+
+        RPP_CALL_DURING_CONSTRUCTION(
+            {
+                observer.set_upstream(disposable);
+            });
+
+        template<typename T>
+        void on_next(T&& v) const
+        {
+            observer.on_next(std::forward<T>(v));
+        }
+
+        void on_error(const std::exception_ptr& err) const
+        {
+            disposable.dispose();
+            selector(err).subscribe(std::move(observer));
+        }
+
+        void on_completed() const
+        {
+            disposable.dispose();
+            observer.on_completed();
+        }
+
+        void set_upstream(const disposable_wrapper& d)
+        {
+            disposable.add(d);
+        }
+
+        bool is_disposed() const { return disposable.is_disposed(); }
+    };
+
+    template<rpp::constraint::decayed_type Selector>
+    struct on_error_resume_next_t : lift_operator<on_error_resume_next_t<Selector>, Selector>
+    {
+        template<rpp::constraint::decayed_type T>
+        struct operator_traits
+        {
+            using selector_observable_result_type =
+                std::decay_t<rpp::utils::extract_observable_type_t<std::invoke_result_t<Selector, std::exception_ptr>>>;
+
+            static_assert(
+                std::is_same_v<selector_observable_result_type, T>,
+                "Selector observable result type is not the same as T");
+
+            using result_type = T;
+
+            template<rpp::constraint::observer_of_type<result_type> TObserver>
+            using observer_strategy = on_error_resume_next_observer_strategy<TObserver, Selector>;
+        };
+
+        template<rpp::details::observables::constraint::disposable_strategy Prev>
+        using updated_disposable_strategy = rpp::details::observables::fixed_disposable_strategy_selector<1>;
```

Comment:
actually it is not fixed<1>. What if Selector result would also set_upstream?

Reply:
To test it, you can return from selector observable with set_upstream in tests

Reply:
Or I can even test by putting nested on_error_resume_next, so I should rather use `atomic_dynamic_disposable_strategy_selector<1>` ?